### DISCUSSION TITLE
onDelete Cascade not working in a one to one relation

### DISCUSSION THREAD

**Discussion by fabriziospadaro** [prisma/prisma#19796] [Q&A] [answered]
I am currently experiencing an issue with deleting users in my Prisma-based application, which appears to be a problem with the deletion cascade for `UserProfiles` and `UserSettings`.

I am using Postgres and prisma client v 4.9.0

I've written a simple test to delete a user and verify that the associated profile and settings are deleted as well:

```js
test("When deleting a user it should also delete 1 profile and 1 setting", async () => {
    const user = await createUser();
    await db.user.delete({ where: { id: user.id } });
    expect(await db.user.count()).toBe(0);
    expect(await db.userProfile.count()).toBe(0);
    expect(await db.userSettings.count()).toBe(0);
  });
```

However, this test is failing. Although the user is deleted successfully, the user's profile and settings remain in the database.
Here are my model schemas:

```prisma
model User {
  id Int @id @default(autoincrement())

  setting UserSettings @relation(fields: [settingId], references: [id], onDelete: Cascade)
  profile UserProfile  @relation(fields: [profileId], references: [id], onDelete: Cascade)

  profileId Int @unique
  settingId Int @unique
}

model UserProfile {
  id Int @id @default(autoincrement())
  User         User?
}

model UserSettings {
  id Int @id @default(autoincrement())
  User           User?
}
```

As you can see, the `User` model is linked to `UserProfile` and `UserSettings` models with the `onDelete: Cascade` option. The intention is to delete the associated `UserProfile` and `UserSettings` whenever a `User` is deleted.

Any guidance or recommendations to troubleshoot and fix this issue would be greatly appreciated.

**Comment by nurul3101** [comment, marked as answer]
Hello @fabriziospadaro 👋 

Can you try using this schema file?

```prisma
model User {
  id Int @id @default(autoincrement())

  setting UserSettings?
  profile UserProfile?
}

model UserProfile {
  id     Int   @id @default(autoincrement())
  User   User? @relation(fields: [userId], references: [id], onDelete: Cascade)
  userId Int?  @unique
}

model UserSettings {
  id     Int   @id @default(autoincrement())
  User   User? @relation(fields: [userId], references: [id], onDelete: Cascade)
  userId Int?  @unique
}
```

Here's the script that I used to verify that `UserProfile` and `UserSettings` are deleted:
```ts
const { PrismaClient } = require('@prisma/client');
const prisma = new PrismaClient();

async function main() {
  // Create a new user with UserProfile and UserSettings
  const newUser = await prisma.user.create({
    data: {
      profile: {
        create: {},
      },
      setting: {
        create: {},
      },
    },
  });
  console.log(`New user: ${newUser.id}`);
  // Delete the user
  await prisma.user.delete({
    where: { id: newUser.id },
  });
  console.log(`Deleted user: ${newUser.id}`);
}

main()
  .catch((e) => {
    throw e;
  })
  .finally(async () => {
    await prisma.$disconnect();
  });
```

The referential actions are defined on the child's side (UserProfile, UserSettings) instead of the parent's side (User).

It would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 
Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Comment by fabriziospadaro** [comment]
Yes, it's working this way, thanks!
